### PR TITLE
[Sprint 2] Feature Header List

### PR DESCRIPTION
**Pull request by tantuan127**
List All Product
Link redmine : https://dev.framgia.com/redmine/issues/113296
Task : Logic Header List All Product

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        void bind(Product product) {
+            mBinding.setViewModel(
+                    new ItemProductViewModel(product, mItemClickListener, mOrderListener));
+            mBinding.executePendingBindings();
+        }
+    }
+
+    private class ContentViewHolder extends RecyclerView.ViewHolder {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: change to
```java
private static class ContentViewHolder extends RecyclerView.ViewHolder
```

File: app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    public void updateData(List<Product> products) {
+        if (products == null) {
+            return;
+        }
+        mProducts.clear();
+        mProducts.addAll(products);
+        notifyDataSetChanged();
+    }
+
+    private class HeaderViewHolder extends RecyclerView.ViewHolder {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: change to
```java
private static class HeaderViewHolder extends RecyclerView.ViewHolder
```

File: app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+/**
+ * Created by levutantuan on 4/21/17.
+ */
+
+public class ListProductAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
+    private List<Product> mProducts;
+    private OnRecyclerViewItemClickListener<Object> mItemClickListener;
+    private static final int HEAD = 0;
+    private static final int CONTENT = 1;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: move ``constant`` variable to the top of this file

File: app/src/main/java/com/framgia/forder/screen/listProduct/ListProductViewModel.java [hunk 4]
```diff
@@ -84,10 +83,10 @@ public void onItemRecyclerViewClick(Object item) {
     }
 
     public LayoutManagers.LayoutManagerFactory getLayoutManager() {
-        return LayoutManagers.grid(mContext.getResources().getInteger(R.integer.item_col));
+        return LayoutManagers.linear();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/ListProductViewModel.java]: remove this method

File: app/src/main/java/com/framgia/forder/screen/mainpage/product/ProductAdapter.java [hunk 5]
```diff
@@ -21,6 +21,7 @@
 
     public ProductAdapter(@NonNull Context context, List<Product> products) {
         super(context);
+
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/mainpage/product/ProductAdapter.java]: remove empty line

File: app/src/main/res/layout/item_header_product.xml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                        android:id="@+id/text_price"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentEnd="true"
+                        android:layout_alignParentRight="true"
+                        android:layout_below="@+id/text_product_name"
+                        android:layout_marginEnd="@dimen/dp_15"
+                        android:layout_marginRight="@dimen/dp_15"
+                        android:ems="5"
+                        android:fontFamily="serif"
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_product.xml]: hard code

File: app/src/main/res/layout/item_list_product.xml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            <TextView
+                android:id="@+id/text_price"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_alignBottom="@+id/image_product"
+                android:layout_alignLeft="@+id/text_shop_name"
+                android:layout_alignStart="@+id/text_shop_name"
+                android:layout_marginBottom="@dimen/dp_2"
+                android:ems="6"
+                android:fontFamily="serif"
```
Comment (daolq3012) [inline app/src/main/res/layout/item_list_product.xml]: hard code

File: app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    public ListProductAdapter(@NonNull Context context, List<Product> products) {
+        super(context);
+        mProducts = new ArrayList<>();
+        mProducts.addAll(products);
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        switch (viewType) {
+            case HEAD:
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: Please declare this constant in this file. & change name to HEADER

File: app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+                ItemHeaderProductBinding itemHeaderProductBinding =
+                        DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
+                                R.layout.item_header_product, parent, false);
+                return new HeaderViewHolder(itemHeaderProductBinding, mItemClickListener,
+                        mOrderListener);
+            case CONTENT:
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: Please declare this constant in this file

Comment (tantuan127) [inline app/src/main/java/com/framgia/forder/screen/listProduct/adapter/ListProductAdapter.java]: Added

File: app/src/main/java/com/framgia/forder/screen/mainpage/product/ProductAdapter.java [hunk 10]
```diff
@@ -63,7 +63,7 @@ public void updateData(List<Product> products) {
         notifyDataSetChanged();
     }
 
-    static class ItemViewHolder extends RecyclerView.ViewHolder {
+    public static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/mainpage/product/ProductAdapter.java]: remove public

**Pull request merged**